numpy/distutils/misc_util.py

```diff
@@ -14,9 +14,15 @@
 import distutils
 from distutils.errors import DistutilsError
 try:
-    from threading import local as tlocal
+    from threading import local as tlocal, Lock
 except ImportError:
-    from dummy_threading import local as tlocal
+    from dummy_threading import local as tlocal, Lock
+
```

Comment:
Yes, append is threadsafe in cpython as the threads can only be changed between bytecode operations. I once saw a list of atomic operations, but don't remember where.